fmf/base.py

```diff
@@ -385,65 +391,75 @@ class describing the environment context. By default, the key
         except KeyError:
             rules = []
 
-        context.case_sensitive = case_sensitive
-
-        # Check and apply each rule
-        for rule in rules:
-
-            # Rule must be a dictionary
-            if not isinstance(rule, dict):
-                raise utils.FormatError("Adjust rule should be a dictionary.")
-
-            original_rule = rule.copy()
-
-            # Missing 'when' means always enabled rule
-            try:
-                condition = rule.pop('when')
-            except KeyError:
-                condition = True
-
-            # The optional 'continue' key should be a bool
-            continue_ = rule.pop('continue', True)
-            if not isinstance(continue_, bool):
-                raise utils.FormatError(
-                    "The 'continue' value should be bool, "
-                    "got '{}'.".format(continue_))
+        additional_rules = copy.deepcopy(additional_rules)
 
-            # The 'because' key is reserved for optional comments (ignored)
-            rule.pop('because', None)
+        # Accept same type as rules from data
+        if additional_rules is None:
+            additional_rules = []
+        elif isinstance(additional_rules, dict):
+            additional_rules = [additional_rules]
 
-            # Apply remaining rule attributes if context matches
-            try:
-                if context.matches(condition):
-                    if decision_callback:
-                        decision_callback(self, original_rule, True)
-
-                    self._merge_special(self.data, rule)
+        context.case_sensitive = case_sensitive
 
-                    # First matching rule wins, skip the rest unless continue
-                    if not continue_:
-                        break
-                else:
+        # Check and apply each rule
+        for set_of_rules in [rules, additional_rules]:
+            for rule in set_of_rules:
```

Comment:
How about doing
```
set_of_rules = rules + additional_rules
for rule in set_of_rules:
```
to avoid the extra indentation?

Reply:
good point, it made rebase much easier.

Reply:
restored, added note. Is there a nicer way? All I could think of is to move the inner loop into own function.

Reply:
Perhaps https://docs.python.org/3.9/library/itertools.html#itertools.chain

Reply:
Doesn't read like it would survive a `break`. I think it's more for maintaining the generator aspect of the iterator.

Would feel very error prone to have any mechanism that would interfere with a language specific like `break`. I think this approach is readable enough. Maybe another approach would be to manipulate the iterators instead of the break, but that would not read well.

Reply:
Agreed. Yeah, this is nicely readable.